lib/rubygems/package/tar_writer.rb

```diff
@@ -234,6 +234,25 @@ def add_file_simple(name, mode, size) # :yields: io
   end
 
   ##
+  # Adds symlink +name+ with permissions +mode+, linking to +target+.
+
+  def add_symlink(name, target, mode)
+    check_closed
+
+    name, prefix = split_name name
+
+    header = Gem::Package::TarHeader.new(:name => name, :mode => mode,
+                                         :size => 0, :typeflag => "2",
+                                         :linkname => target,
+                                         :prefix => prefix,
+                                         :mtime => Time.now).to_s
+
+    @io.write header
+
+    self
+  end
```

Comment:
This can mirror [`test_add_file`](https://github.com/rubygems/rubygems/blob/master/test/rubygems/test_gem_package_tar_writer.rb#L22-L31)

It might need more tests for digests and signers, but I don't remember off the top of my head so I can backfill those for you after merge.